### ISSUE TITLE
Overfitting problem?

### ISSUE THREAD

**Issue by izxi** [izxi/Learning#14]
If you meet an overfitting problem in you model, which of the following methods can help you improve the model?

Select one or more:

- [x]  A. Add new domain-specific features
- [x]  B. Increase the amount of training data
- [x]  C. Increase the amount of regularization used
- [x]  D. Cross Validation

**Comment by izxi** [comment]
[overfit](https://www.autonlab.org/_media/tutorials/overfit10.pdf)

**Comment by izxi** [comment]
[overfitting-in-machine-learning](https://elitedatascience.com/overfitting-in-machine-learning)

Cross-validation
Cross-validation is a powerful preventative measure against overfitting.
In standard k-fold cross-validation, we partition the data into k subsets, called folds. Then, we iteratively train the algorithm on k-1 folds while using the remaining fold as the test set (called the “holdout fold”).

Train with more data
It won’t work everytime, but training with more data can help algorithms detect the signal better.

Remove features
Some algorithms have built-in feature selection.
For those that don’t, you can manually improve their generalizability by removing irrelevant input features.

Early stopping
When you’re training a learning algorithm iteratively, you can measure how well each iteration of the model performs.

Regularization
Regularization refers to a broad range of techniques for artificially forcing your model to be simpler.

Ensembling
Ensembles are machine learning methods for combining predictions from multiple separate models. There are a few different methods for ensembling, but the two most common are:
Bagging attempts to reduce the chance overfitting complex models.

- It trains a large number of "strong" learners in parallel.
- A strong learner is a model that's relatively unconstrained.
- Bagging then combines all the strong learners together in order to "smooth out" their predictions.

**Comment by izxi** [comment]
[DSL](https://www.slideshare.net/orendon/medellinrb-domain-specific-languages-in-ruby-march-2015)

**Comment by johnnieng** [comment]
Not A, B, C, D
